### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by MoRaad97**
## Project requirements

- Create a class called Solver.
- Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
- When N is divisible by 3, return "fizz".
- When N is divisible by 5, return "buzz".
- When N is divisible by 3 and 5, return "fizzbuzz".
- Any other case, return N as a string (e.g. say N is 7 then return "7").

<hr>

### In This PR

1. Add Tests for all the functions using TDD concept.
2. Create the class Solver.
3. Add the following methods to the class "Solver":
    - factorial.
    - reverse.
    - fizzbuzz.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+describe Solver do
+  solver = Solver.new
+  context 'factorial Method testing' do
+    it 'should return 1 if the the value = 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'should return error if the value < 0' do
+      expect { solver.factorial(-3) }.to raise_error(ArgumentError)
```
Comment (GzimAsani) [inline spec/solver_spec.rb]: - [x] Add the correct error message, it should be the exception one:
`expect { resolver.factorial(-3) }.to raise_exception`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Profsain and @MoRaad97           :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with tests +1: 
 
 ## Required Changes 

- [x] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (MoRaad97) [reply spec/solver_spec.rb]: Thanks, @GzimAsani  will start working on it now.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi team, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**